fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneIndexFailureTest.java

```diff
@@ -273,106 +272,122 @@ void repartitionAndMerge(boolean useLegacyAsyncToSync) throws IOException {
                     allIds.stream()
                             .skip(190)
-                            .map(idLong -> Tuple.from(docGroupFieldValue, idLong))
                             .collect(Collectors.toSet()),
-                    "text:propose");
+                    LuceneIndexTestDataModel.PARENT_SEARCH_TERM,
+                    store);
             for (int i = 1; i < 20; i++) {
                 // 0 should have the newest
                 // everyone else should increase
-                validateDocsInPartition(index, i, groupingKey,
+                dataModel.validateDocsInPartition(dataModel.index, i, groupingKey,
                         allIds.stream().skip((i - 1) * partitionSize)
                                 .limit(partitionSize)
-                                .map(idLong -> Tuple.from(docGroupFieldValue, idLong))
                                 .collect(Collectors.toSet()),
-                        "text:propose");
+                        LuceneIndexTestDataModel.PARENT_SEARCH_TERM,
+                        store);
             }
-            List<LucenePartitionInfoProto.LucenePartitionInfo> partitionInfos = getPartitionMeta(index,
-                    groupingKey, contextProps, schemaSetup);
+        }
+        try (FDBRecordContext context = openContext(contextProps)) {
+            FDBRecordStore store = dataModel.createOrOpenRecordStore(context);
+            List<LucenePartitionInfoProto.LucenePartitionInfo> partitionInfos = LuceneIndexTestUtils.getPartitionMeta(dataModel.index, groupingKey, store);
             assertEquals(partitionCount, partitionInfos.size());
         }
     }
 
     @ParameterizedTest
     @BooleanSource
     void optimizedPartitionInsertionTest(boolean useLegacyAsyncToSync) throws IOException {
-        Index index = COMPLEX_PARTITIONED;
+        long seed = 657368233L;
         Tuple groupingKey = Tuple.from(1L);
         final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
-                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 6)
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 10)
+                .addProp(LuceneRecordContextProperties.LUCENE_MERGE_SEGMENTS_PER_TIER, (double)2)
                 .addProp(LuceneRecordContextProperties.LUCENE_USE_LEGACY_ASYNC_TO_SYNC, useLegacyAsyncToSync)
                 .build();
 
         final int totalDocCount = 10; // configured index's highwater mark
-        Consumer<FDBRecordContext> schemaSetup = context -> rebuildIndexMetaData(context, COMPLEX_DOC, index);
-        long docGroupFieldValue = groupingKey.isEmpty() ? 0L : groupingKey.getLong(0);
+
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilderWithRegistry, pathManager)
+                .setIsGrouped(true)
+                .setPartitionHighWatermark(10)
```

Comment:
Based on the comment above, should this be:
```suggestion
                .setPartitionHighWatermark(totalDocCount)
```